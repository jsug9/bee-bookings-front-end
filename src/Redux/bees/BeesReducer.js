/* eslint-disable no-param-reassign */
import axios from 'axios';
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const GET_BEES = 'Bees/GET_BEES';
const GET_BEE_DETAILS = 'Bees/GET_BEE_DETAILS';
const ADD_BEE = 'Bees/ADD_BEE';
const DELETE_BEE = 'Bees/DELETE_BEE';

const getBeesEndpoint = 'https://bee-store.herokuapp.com/api/v1/items';

const initialState = {
  allBees: [],
  beeDetails: {},
  isLoading: false,
};

export const getAllBees = createAsyncThunk(GET_BEES, async () => {
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

export const getBeeDetails = createAsyncThunk(GET_BEE_DETAILS, async (id) => {
  const { data } = await axios.get(`${getBeesEndpoint}/${id}`);
  return data;
});

export const addBee = createAsyncThunk(ADD_BEE, async (bee) => {
  axios.post(getBeesEndpoint, bee);
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

export const deleteBee = createAsyncThunk(DELETE_BEE, async (beeId) => {
  axios.delete(`${getBeesEndpoint}/${beeId}`);
  const { data } = await axios.get(getBeesEndpoint);
  return data;
});

const beesSlice = createSlice({
  name: 'Bees',
  initialState,
  reducers: {},
  extraReducers: {
    [getAllBees.fulfilled]: (state, action) => {
      state.allBees = action.payload.sort((a, b) => a.id - b.id);
      state.isLoading = false;
    },
    [getAllBees.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [getAllBees.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
    [getBeeDetails.fulfilled]: (state, action) => {
      state.beeDetails = action.payload;
      state.isLoading = false;
    },
    [getBeeDetails.rejected]: (state) => {
      state.beeDetails = {};
      state.isLoading = false;
    },
    [getBeeDetails.pending]: (state) => {
      state.beeDetails = {};
      state.isLoading = true;
    },
    [addBee.fulfilled]: (state, action) => {
      state.allBees = action.payload;
      state.isLoading = false;
    },
    [addBee.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [addBee.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
    [deleteBee.fulfilled]: (state, action) => {
      state.allBees = action.payload;
      state.isLoading = false;
    },
    [deleteBee.rejected]: (state) => {
      state.allBees = [];
      state.isLoading = false;
    },
    [deleteBee.pending]: (state) => {
      state.allBees = [];
      state.isLoading = true;
    },
  },
});

export default beesSlice.reducer;
