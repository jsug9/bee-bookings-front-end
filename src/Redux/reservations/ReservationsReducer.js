/* eslint-disable no-param-reassign */
import axios from 'axios';
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const getReservationsUrl = 'https://bee-store.herokuapp.com/api/v1/bookings/user';
const addReservationUrl = 'https://bee-store.herokuapp.com/api/v1/bookings';

const ADD_RESERVATION = 'Reservations/ADD_RESERVATION';
const DELETE_RESERVATION = 'Reservations/DELETE_RESERVATION';
const GET_RESERVATIONS = 'Reservations/GET_RESERVATIONS';

export const getReservations = createAsyncThunk(GET_RESERVATIONS, async (userId) => {
  const { data } = await axios.post(getReservationsUrl, { user_id: userId });
  return data;
});

export const addReservation = createAsyncThunk(ADD_RESERVATION, async (reservation) => {
  axios.post(addReservationUrl, reservation);
  const { data } = await axios.post(getReservationsUrl, { user_id: reservation.user_id });
  return data;
});

export const deleteReservation = createAsyncThunk(DELETE_RESERVATION,
  async (reservationId) => {
    try {
      axios.delete(`${addReservationUrl}/${reservationId}`);
    } catch (error) {
      return error;
    }
    return reservationId;
  });

const initialState = {
  allReservations: [],
  isLoading: false,
};

const reservationsSlice = createSlice({
  name: 'reservations',
  initialState,
  reducers: {
    clearReservations: (state) => {
      state.allReservations = [];
      state.isLoading = false;
    },
  },
  extraReducers: {
    [getReservations.fulfilled]: (state, action) => {
      state.allReservations = action.payload;
      state.isLoading = false;
    },
    [getReservations.rejected]: (state) => {
      state.allReservations = [];
      state.isLoading = false;
    },
    [getReservations.pending]: (state) => {
      state.isLoading = true;
    },
    [addReservation.fulfilled]: (state, action) => {
      state.allReservations = action.payload;
      state.isLoading = false;
    },
    [addReservation.rejected]: (state) => {
      state.allReservations = [];
      state.isLoading = false;
    },
    [addReservation.pending]: (state) => {
      state.isLoading = true;
    },
    [deleteReservation.fulfilled]: (state, action) => {
      state.allReservations = state.allReservations.filter(
        (reservation) => reservation.id !== action.payload,
      );
      state.isLoading = false;
    },
    [deleteReservation.rejected]: (state) => {
      state.allReservations = [];
      state.isLoading = false;
    },
    [deleteReservation.pending]: (state) => {
      state.isLoading = true;
    },
  },
});

export const { clearReservations } = reservationsSlice.actions;
export default reservationsSlice.reducer;
