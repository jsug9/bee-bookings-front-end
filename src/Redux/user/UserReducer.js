/* eslint-disable no-param-reassign */
import axios from 'axios';
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

const signInEndpoint = 'https://bee-store.herokuapp.com/api/v1/signin';
const signUpEndpoint = 'https://bee-store.herokuapp.com/api/v1/signup';

export const signIn = createAsyncThunk('user/signIn', async (username) => {
  const userParams = { user: { username } };
  const { data } = await axios.post(signInEndpoint, userParams);
  return data;
});

export const sum = () => 'from user reducer';

export const signUp = createAsyncThunk('user/signUp', async (username) => {
  const userParams = { user: { username } };
  const { data } = await axios.post(signUpEndpoint, userParams);
  return data;
});

const initialState = {
  username: null,
  userId: null,
  isLoading: false,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logOutUser: (state) => {
      state.username = null;
      state.userId = null;
      state.createdUser = null;
    },
  },
  extraReducers: {
    [signIn.pending]: (state) => {
      state.isLoading = true;
    },
    [signIn.fulfilled]: (state, action) => {
      state.username = action.payload.username;
      state.userId = action.payload.user_id;
      state.isLoading = false;
    },
    [signIn.rejected]: (state) => {
      state.isLoading = false;
    },
    [signUp.pending]: (state) => {
      state.isLoading = true;
    },
    [signUp.fulfilled]: (state, action) => {
      state.username = action.payload.username;
      state.userId = action.payload.user_id;
      state.isLoading = false;
      state.createdUser = true;
    },
    [signUp.rejected]: (state) => {
      state.isLoading = false;
    },
  },
});

export const { logOutUser } = userSlice.actions;
export default userSlice.reducer;
