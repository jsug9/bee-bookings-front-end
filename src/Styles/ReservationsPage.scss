.reservations {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;

  h1 {
    text-align: center;
  }
}

.center {
  display: flex;
  justify-content: center;

  h1 {
    text-align: center;
  }
}
