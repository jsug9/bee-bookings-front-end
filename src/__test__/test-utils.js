/* eslint-disable react/prop-types */
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import setupStore from '../Redux/testStore';

function renderWithProviders(
  ui,
  {
    preloadedState,
    store = setupStore(preloadedState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Router><Provider store={store}>{children}</Provider></Router>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export default renderWithProviders;
