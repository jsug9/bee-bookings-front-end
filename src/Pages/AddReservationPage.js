/* eslint-disable react/jsx-props-no-spreading */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  Select, MenuItem, InputLabel, FormControl, TextField, Button,
} from '@mui/material';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { MobileDatePicker } from '@mui/x-date-pickers/MobileDatePicker';
import LibraryAddIcon from '@mui/icons-material/LibraryAdd';
import { useLocation } from 'react-router-dom';
import { addReservation } from '../Redux/reservations/ReservationsReducer';
import { getAllBees } from '../Redux/bees/BeesReducer';
import PleaseLogin from '../Components/PleaseLogin';
import '../Styles/BeeForm.scss';
import '../Styles/ReservationForm.scss';

const AddReservationPage = () => {
  const location = useLocation();
  const [date, setDate] = useState(new Date());
  const [beeId, setBeeId] = useState('');
  const [city, setCity] = useState('');
  const dispatch = useDispatch();
  const userId = useSelector((state) => state.user.userId);
  const bees = useSelector((state) => state.bees.allBees);

  useEffect(() => {
    if (location.state) {
      const { bee } = location.state;
      setBeeId(bee.id);
    }
    dispatch(getAllBees());
  }, []);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (city.trim()) {
      const object = {
        date: date.toString(),
        user_id: userId,
        item_id: beeId,
        city: city.trim(),
      };
      dispatch(addReservation(object));
      setDate(new Date());
      setBeeId('');
      setCity('');
    }
  };

  return (
    <div className="reservation-form-container">
      { userId
      && (
      <form className="bee-form reservation-form" method="post" onSubmit={handleSubmit}>
        <h1 style={{ marginBottom: '30px' }}>Add Reservation Page</h1>
        <FormControl>
          <InputLabel id="bee-label">Bee</InputLabel>
          <Select required labelId="bee-label" label="Bee" style={{ backgroundColor: 'white', marginBottom: '20px', borderRadius: '4px' }} value={beeId} onChange={(e) => { setBeeId(e.target.value); }}>
            {bees.map((bee) => (
              <MenuItem value={bee.id} key={bee.id}>
                {bee.name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
        <LocalizationProvider dateAdapter={AdapterDateFns}>
          <MobileDatePicker
            label="Date"
            inputFormat="MM/dd/yyyy"
            value={date}
            required
            onChange={(newValue) => setDate(newValue)}
            renderInput={(params) => <TextField sx={{ backgroundColor: 'white', borderRadius: '4px' }} {...params} />}
          />
        </LocalizationProvider>
        <TextField id="city" label="City" variant="outlined" style={{ borderRadius: '4px', backgroundColor: 'white', marginTop: '20px' }} required value={city} onChange={(e) => setCity(e.target.value)} />
        <Button
          type="submit"
          variant="contained"
          color="success"
          startIcon={<LibraryAddIcon />}
          sx={{
            fontWeight: 'bold', marginTop: '20px', width: '300px', alignSelf: 'center',
          }}
        >
          Book bee
        </Button>
      </form>
      )}
      {!userId
      && (
        <PleaseLogin message="Add a Reservation" />
      )}
    </div>
  );
};

export default AddReservationPage;
