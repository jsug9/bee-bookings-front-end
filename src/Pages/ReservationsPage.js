import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import BasicTable from '../Components/Table';
import '../Styles/ReservationsPage.scss';
import { getReservations } from '../Redux/reservations/ReservationsReducer';
import { getAllBees } from '../Redux/bees/BeesReducer';
import PleaseLogin from '../Components/PleaseLogin';

const emptyReservationsNotice = () => (
  <div className="center">
    <h1>
      No Bees Booked 🐝
    </h1>
  </div>
);

const ReservationsPage = () => {
  const user = useSelector((state) => state.user);
  const bees = useSelector((state) => state.bees.allBees);
  const reservations = useSelector((state) => state.reservations.allReservations);

  const dispatch = useDispatch();

  const [isEmpty, setIsEmpty] = useState(true);

  useEffect(() => {
    // checks if bees has been fetched. If not, fetches bees from the database
    if (bees.length === 0) {
      dispatch(getAllBees());
    }
    dispatch(getReservations(user.userId));
  }, [user]);

  useEffect(() => {
    // checks if bees has been fetched. If not, fetches bees from the database
    if (reservations.length > 0) {
      setIsEmpty(false);
    } else {
      setIsEmpty(true);
    }
  }, [reservations]);

  if (user.userId && isEmpty) {
    return emptyReservationsNotice();
  }

  if (user.userId) {
    return (
      <div className="reservations">
        <h1 className="bookTableHeader">
          {'Bees booked by '}
          {user?.username}
        </h1>
        <BasicTable />
      </div>
    );
  }

  return (<PleaseLogin message="View Your Reservations" />);
};

export default ReservationsPage;
