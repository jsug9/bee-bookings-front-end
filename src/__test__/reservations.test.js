import { act } from 'react-dom/test-utils';
import ReservationsPage from '../Pages/ReservationsPage';
import AddReservationPage from '../Pages/AddReservationPage';
import { signIn } from '../Redux/user/UserReducer';
import { getAllBees } from '../Redux/bees/BeesReducer';
import { addReservation, deleteReservation, getReservations } from '../Redux/reservations/ReservationsReducer';
import renderWithProviders, { screen } from './test-utils';
import setupStore from '../Redux/testStore';
import server from '../mswMocks/server';

// Establish API mocking before all tests.
beforeAll(() => server.listen());
// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());
// Clean up after the tests are finished.
afterAll(() => server.close());

it('renders reservations to the page', () => {
  const tree = renderWithProviders(
    <ReservationsPage />,
  );
  expect(tree).toMatchSnapshot();
});

it('renders add reservations form to the page', () => {
  const tree = renderWithProviders(
    <AddReservationPage />,
  );
  expect(tree).toMatchSnapshot();
});

it('doesn\'t show the add reservations form if the user is not logged in', async () => {
  const store = setupStore();
  //   await store.dispatch(signIn('AaronIsCool'));

  act(() => {
    renderWithProviders(
      <AddReservationPage />, { store },
    );
  }).then(() => {
    expect(screen.queryByText(/Add Reservation Page/)).not.toBeInTheDocument();
  });
});

it('doesn\'t show the reservations page if the user is not logged in', async () => {
  const store = setupStore();

  act(() => {
    renderWithProviders(
      <ReservationsPage />, { store },
    );
  }).then(() => {
    expect(screen.queryByText(/Bees booked/)).not.toBeInTheDocument();
  });
});

it('shows the add reservations form if the user is logged in', async () => {
  const store = setupStore();
  await store.dispatch(signIn('AaronIsCool'));

  act(() => {
    renderWithProviders(
      <AddReservationPage />, { store },
    );
  });

  expect(screen.getByText('Add Reservation Page')).toBeInTheDocument();
});

it('shows the reservations page if the user is logged in', async () => {
  const store = setupStore();
  await store.dispatch(signIn('AaronIsCool'));

  act(() => {
    renderWithProviders(
      <ReservationsPage />, { store },
    );
  });

  expect(screen.getByText('No Bees Booked 🐝')).toBeInTheDocument();
});

it('shows the reservations if the user is logged in', async () => {
  const store = setupStore();
  await store.dispatch(signIn('emyrue'));
  await store.dispatch(getAllBees());
  await store.dispatch(getReservations(5));

  act(() => {
    renderWithProviders(
      <ReservationsPage />, { store },
    );
  });

  expect(screen.getByText('Bees booked by AaronIsCool')).toBeInTheDocument();
});

it('shows the added reservations', async () => {
  const store = setupStore();
  await store.dispatch(signIn('emyrue'));
  await store.dispatch(getAllBees());
  await store.dispatch(addReservation({
    item_id: 1,
    user_id: 5,
    date: '2020-01-01',
    city: 'New York',
  }));

  act(() => {
    renderWithProviders(
      <ReservationsPage />, { store },
    );
  });

  expect(store.getState().reservations.allReservations).not.toHaveLength(0);
});

it('deletes a reservation', async () => {
  const store = setupStore();
  await store.dispatch(signIn('emyrue'));
  await store.dispatch(getAllBees());

  await act(async () => {
    renderWithProviders(
      <ReservationsPage />, { store },
    );

    await store.dispatch(deleteReservation(37));
  });

  expect(store.getState().reservations.allReservations).not.toHaveLength(0);
});
