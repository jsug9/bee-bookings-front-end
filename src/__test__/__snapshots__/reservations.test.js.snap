// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add reservations form to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reservation-form-container"
      >
        <div
          class="please-log-in-container"
        >
          <h1
            class="please-log-in"
          >
            Please Log In to
             
            Add a Reservation
          </h1>
          <div
            class="please-log-in-button-container"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LoginIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                  />
                </svg>
              </span>
              Log In
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="PersonAddIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </span>
              Sign Up
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reservation-form-container"
    >
      <div
        class="please-log-in-container"
      >
        <h1
          class="please-log-in"
        >
          Please Log In to
           
          Add a Reservation
        </h1>
        <div
          class="please-log-in-button-container"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LoginIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                />
              </svg>
            </span>
            Log In
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </span>
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`renders reservations to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="please-log-in-container"
      >
        <h1
          class="please-log-in"
        >
          Please Log In to
           
          View Your Reservations
        </h1>
        <div
          class="please-log-in-button-container"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LoginIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                />
              </svg>
            </span>
            Log In
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonAddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </span>
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="please-log-in-container"
    >
      <h1
        class="please-log-in"
      >
        Please Log In to
         
        View Your Reservations
      </h1>
      <div
        class="please-log-in-button-container"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LoginIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
              />
            </svg>
          </span>
          Log In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-6zbjo5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonAddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </span>
          Sign Up
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
